numpy/_core/tests/test_dtype.py

```diff
@@ -436,6 +443,14 @@ def test_subarray_list(self, obj, dtype, expected):
 
         assert_array_equal(res, expected)
 
+    def test_parenthesized_single_number(self):
+        with pytest.raises(TypeError, match="not understood"):
+            np.dtype("(2)f4")
+
+        with pytest.warns(DeprecationWarning,
+                          match="parenthesized single number"):
+            np.dtype("(2)f4,")
```

Comment:
Would be nice as a Deprecation test case, mainly because that also raises the error, thus covering the error path.  But, it does also look all good anyway.

Reply:
I added a deprecation test case and a comment here. Let me know if this is what you had in mind.